### PR TITLE
Test inventory order in a unit test

### PR DESCRIPTION
**Pull request by 14rcole**

### CODE REVIEW

**Comment by p3ck** [comment]
Oops, I spoke too soon.  If you dig into the failure it's because it can't open a file.  getting close though.

**Comment by 14rcole** [comment]
[test]

**Comment by p3ck** [comment]
@14rcole it passed the unit test, but the code hasn't been fixed yet right?
        "linchpin/tests/InventoryFilters/test_GenericInventory_pass.py::test_output_order PASSED [ 49%]",

File: linchpin/tests/InventoryFilters/test_GenericInventory_pass.py [hunk 1]
```diff
@@ -5,6 +5,8 @@
 import os
 import json
 import yaml
+import json
```
Comment (samvarankashyap) [inline linchpin/tests/InventoryFilters/test_GenericInventory_pass.py]: @14rcole  Duplicate import

**Comment by 14rcole** [comment]
@p3ck I only added the test.  The code seems to be working, at least this is the expected output here.

One thing worth questioning is how similar do these topologies need to be?  I compared the files as if they needed to be identical, but really, each section just needs to match.  The sections themselves could be reordered and be fine, right?  If so, I should modify the verification to read in the expected inventory to the ConfigParser and compare sections from there

**Comment by samvarankashyap** [comment]
> @p3ck I only added the test. The code seems to be working, at least this is the expected output here.
> 
> One thing worth questioning is how similar do these topologies need to be? I compared the files as if they needed to be identical, but really, each section just needs to match. The sections themselves could be reordered and be fine, right? If so, I should modify the verification to read in the expected inventory to the ConfigParser and compare sections from there

@14rcole A simpler test would be a pattern match on a file using regex for sections, instances should output the identical sections every time its ran.

**Comment by p3ck** [comment]
My suggestion would be to take the example topology here and run it for real.  See if you get the same results in the real world.

Also - Don't we have a reproducer from issue from #769 ?

I really want to see the problem reproduced with this test case.

**Comment by 14rcole** [comment]
#769 mysteriously stopped reproducing.  I don't know if another PR accidentally fixed it. I left the issue open because of [this](https://github.com/CentOS-PaaS-SIG/linchpin/issues/769#issuecomment-434792907), which I meant to bring up at the linchpin devel meeting or not.  I'm not sure if that needs to be fixed.  Other than that, the example topology did produce the expected inventory (albeit with the sections in a different order - I had to reorder the expected output to match)

**Comment by p3ck** [comment]
ok - I agree that the order of the hosts in each section shouldn't matter, as long as they end up in the correct sections.

@samvarankashyap Can you think of any other combinations that could be a problem?  I know in the beaker provider I can provide multiple recipesets:

yaml```
        recipesets:
          - distro: RHEL-7.4
            arch: x86_64
            variant: Server
            count: 1
            name: rhel75-1
            keyvalue:
              - DISKSPACE>=100000
            bkr_data: lvm-on
            hostrequires:
              - tag: arch
                op: "="
                value: "X86_64"
              - tag: memory
                op: ">="
                value: "1000"
          - distro: RHEL-7.5
            arch: x86_64
            variant: Server
            count: 1
            name: rhel75-2
            keyvalue:
              - DISKSPACE>=100000
```

Can we add this use case into the test?

**Comment by 14rcole** [comment]
I added this to the test.  As it turns out, we do need to compare the files verbatim. ConfigParser can't read the inventory files it's generating because the [host_group:children] sections have keys without values

**Comment by p3ck** [comment]
@14rcole did you push the latest version of this patch?  You stated you added my example to the test but I don't see it.

**Comment by 14rcole** [comment]
@p3ck I just pushed up the most recent changes.  I spent a while wrestling with beaker

**Comment by 14rcole** [comment]
[merge]

**Pull request merged**